panther_battery/src/dual_battery_publisher.cpp

```diff
@@ -153,4 +154,30 @@ void DualBatteryPublisher::MergeBatteryPowerSupplyHealth(
   }
 }
 
+void DualBatteryPublisher::DiagnoseBattery(diagnostic_updater::DiagnosticStatusWrapper & status)
+{
+  std::vector<diagnostic_msgs::msg::KeyValue> key_values;
+  unsigned char error_level{diagnostic_updater::DiagnosticStatusWrapper::OK};
+  std::string message{"Battery has no error messages"};
+
+  if (battery_1_->HasErrorMsg() || battery_2_->HasErrorMsg()) {
+    error_level = diagnostic_updater::DiagnosticStatusWrapper::ERROR;
+    message = "Battery has error";
+
+    diagnostic_msgs::msg::KeyValue battery_1_kv;
+    battery_1_kv.key = "Error message: battery 1";
+    battery_1_kv.value = battery_1_->GetErrorMsg();
+
+    diagnostic_msgs::msg::KeyValue battery_2_kv;
+    battery_2_kv.key = "Error message: battery 2";
+    battery_2_kv.value = battery_2_->GetErrorMsg();
+
+    key_values.push_back(battery_1_kv);
+    key_values.push_back(battery_2_kv);
+  }
```

Comment:
There is LogError method that checks similar conditions. I wonder if we could move contents of LogError() in here and do: `diagnostic_updater_->force_update()` to achieve the same result. This would however depend on the way how force_update works - dose it reset updater publisher timer. The idea is if publish diagnostic in 10Hz and call force_update 10Hz. Will the diagnostic be publishing 20 messages every second? 
Soooo we don't wont to publish to much diagnostic messages but if force_update resets publisher timer  then it may be usfull to move LogErrors content in here.

Reply:
We can move the stuff from `LogError` to the diagnostic routine. Together with the `throttle` logging macro, it'll not spam us with too frequent messages in the terminal.

Done ✅

Reply:
As we discussed, we go pack to using LogError